.cascade-code/Chapman.edu/test-section/tatwell/fast-omni-nav/omni_nav.vtl

```diff
@@ -0,0 +1,514 @@
+  ## drop-down menu with multiple links.
+  #set ( $hideSettingSelector = "dynamic-metadata[1][value='Yes'] or system-page[name = 'index']/dynamic-metadata[1][value='Yes']" )
+  #set ( $metaHideSetting = $_XPathTool.selectNodes($siblingNode, $hideSettingSelector) )
+  #set ( $isVisible = $metaHideSetting == [false] )
+
+  ## If hidden, continue. But velocity doesn't have a continue operator:
+  ## https://stackoverflow.com/questions/7530019/apache-velocity-foreach-loop-continue
+  #if ( $isVisible )
+    #set ( $nodeType = $siblingNode.getName() )
+    #set ( $link = $siblingNode.getChild("link").value )
+
+    #if ( $nodeType == 'system-page' )
+      #set ( $displayName = $_EscapeTool.xml($siblingNode.getChild("display-name").value) )
+      <li><a href="$link">$displayName</a></li>
+
+    #elseif ( $nodeType == 'system-folder' )
+      #set ( $folderIndexPage = $_XPathTool.selectSingleNode($siblingNode, "system-page[name = 'index']") )
+      #set ( $displayName = $_EscapeTool.xml($folderIndexPage.getChild("display-name").value) )
+      #set ( $children = $_XPathTool.selectNodes($siblingNode, "system-page[name != 'index'] | system-folder") )
+
+      ## Count hidden children to see if we need to show toggle button.
+      #set ( $hiddenChildCounter = 0 )
+      #set ( $hiddenChildSelector = "dynamic-metadata[1][value='Yes'] | system-folder/dynamic-metadata/value" )
+      #foreach ( $child in $children )
+        #if ( $_XPathTool.selectSingleNode($child, $hiddenChildSelector) )
+          #set ( $hiddenChildCounter = $hiddenChildCounter + 1 )
+        #end
+      #end
+      #set ( $hasVisibleChildren = $children.size() > 0 && $hiddenChildCounter < $children.size() )
+
+      <li>
+        <a href="${link}/index">$displayName</a>
+
+        #if ( $hasVisibleChildren )
+          <span class="toggle"><span>+</span></span>
+          <ul>
+            #foreach ( $child in $children )
+              #set ( $metaHideSettingSelector = "dynamic-metadata[1][value='Yes'] or system-page[name = 'index']/dynamic-metadata[1][value='Yes']" )
+              #set ( $metaHideChildSetting = $_XPathTool.selectNodes($child, $metaHideSettingSelector) )
+              #set ( $childIsVisible = $metaHideChildSetting == [false] )
+
+              #if ( $childIsVisible )
+                #set ( $childFileType = $child.getName() )
+                #set ( $childLink = $child.getChild("link").value )
+                #set ( $childDisplayName = $_EscapeTool.xml($child.getChild("display-name").value) )
+
+                #if ( $childFileType == "system-page" )
+                  <li><a href="${childLink}">$childDisplayName</a></li>
+                #elseif ( $childFileType == "system-folder" )
+                  <li><a href="${childLink}/index">$childDisplayName</a></li>
+                #end
+              #end
+            #end
+          </ul>
+        #end
+      </li>
+    #end
+  #end
+#end
+
+
+##
+## Main OmniNav HTML Block
```

Comment:
I like this "main block" pattern with most functionality encapsulated in macros.

Reply:
👍